runtime/src/bank.rs

```diff
@@ -4414,6 +4427,28 @@ impl Bank {
             .is_active(&feature_set::stake_program_v2::id())
     }
 
+    pub fn simpler_capitalization_enabled(&self) -> bool {
+        self.simpler_capitalization_enabled_at_genesis()
+            || self
+                .feature_set
+                .is_active(&feature_set::simpler_capitalization::id())
+    }
+
+    fn simpler_capitalization_enabled_at_genesis(&self) -> bool {
+        // genesis builtin initialization codepath is called even before the initial
+        // feature activation, so we need to peek this flag at very early bank
+        // initialization phase for the development genesis case
+        if let Some(account) = self.get_account(&feature_set::simpler_capitalization::id()) {
+            if let Some(feature) = feature::from_account(&account) {
+                if feature.activated_at == Some(0) {
+                    return true;
```

Comment:
I think this is pretty safe, not having any possibility of falsely activating this feature on testnet/mainnet-beta.